pkg/container-utils/cri/cri.go

```diff
@@ -330,16 +323,44 @@ func containerStatusStateToRuntimeClientState(containerStatusState runtime.Conta
 	return
 }
 
-func CRIContainerToContainerData(runtimeName string, container *runtime.Container) *runtimeclient.ContainerData {
+// CRIContainer is an interface that contains the methods required to get
+// the information of a container from the responses of the CRI. In particular,
+// from runtime.ContainerStatus and runtime.Container.
+type CRIContainer interface {
+	GetId() string
+	GetState() runtime.ContainerState
+	GetMetadata() *runtime.ContainerMetadata
+	GetLabels() map[string]string
+}
+
+func CRIContainerToContainerData(runtimeName types.RuntimeName, container CRIContainer) *runtimeclient.ContainerData {
+	containerMetadata := container.GetMetadata()
+
 	containerData := &runtimeclient.ContainerData{
-		ID:      container.Id,
-		Name:    strings.TrimPrefix(container.GetMetadata().Name, "/"),
-		State:   containerStatusStateToRuntimeClientState(container.GetState()),
-		Runtime: runtimeName,
+		Runtime: runtimeclient.RuntimeContainerData{
+			BasicRuntimeMetadata: types.BasicRuntimeMetadata{
+				ContainerID:   container.GetId(),
+				ContainerName: strings.TrimPrefix(containerMetadata.GetName(), "/"),
+				RuntimeName:   runtimeName,
+			},
+			State: containerStatusStateToRuntimeClientState(container.GetState()),
+		},
 	}
 
 	// Fill K8S information.
-	runtimeclient.EnrichWithK8sMetadata(containerData, container.Labels)
+	runtimeclient.EnrichWithK8sMetadata(containerData, container.GetLabels())
+
+	// CRI-O does not use the same container name of Kubernetes as containerd.
+	// Instead, it uses a composed name as Docker does, but such name is not
+	// available in the container metadata.
+	if runtimeName == types.RuntimeNameCrio {
```

Comment:
How about using annotation `io.kubernetes.cri-o.Name` for it? You can see it in [workflow run](https://github.com/inspektor-gadget/inspektor-gadget/actions/runs/5486300137/jobs/9996327362#step:7:27193) here. I see it listed as future PR so fine handling it later.

Reply:
Yes, I will do that in the third TODO I left for the future PR. I'm also getting it from the container-hook but it is not valid for initial containers. In general, we need to document and test container enrichers as we need to know exactly what elements is each enricher providing. I'll try to work on that on this sprint.

Reply:
> we need to document and test container enrichers as we need to know exactly what elements is each enricher providing.

Totally agree, I know each runtime has subtle differences that we have to keep in mind so would be great to document it. 

Also, as part of this [issue](https://github.com/inspektor-gadget/inspektor-gadget/issues/1613) I [experimented](https://github.com/inspektor-gadget/inspektor-gadget/commit/e2c6158273116cd1a5bfa94d205ee23847220ccf) with `OCIConfig` for initial containers and it seemed to be [working fine](https://github.com/inspektor-gadget/inspektor-gadget/issues/1613#issuecomment-1540495562). Can help us unify use-cases for initial vs new containers.

Reply:
That's great! Unifying initial vs new containers will help, but in general, we need an internal documentation saying how we are getting the information for each runtime and whether it works for both initial and new containers.